public/sw.js

```diff
@@ -22,8 +22,11 @@ self.addEventListener("push", (e) => {
   );
 });
 
-// self.addEventListener("notificationclick", (e) => {
-//   clients.openWindow("http://localhost:3000/main" + e.notification.data?.url);
-// });
+self.addEventListener("notificationclick", (e) => {
+  //TODO: get host from env
+  clients.openWindow(
+    "http://localhost:3000/main" + e.notification.data?.convId
```

Comment:
`http://localhost:3000` should not be hardcoded. You need to get current host from env